crates/rpc/rpc/src/eth/pubsub.rs

```diff
@@ -146,14 +146,20 @@ where
                 match params {
                     Params::Bool(true) => {
                         // full transaction objects requested
-                        let stream = pubsub.full_pending_transaction_stream().map(|tx| {
-                            EthSubscriptionResult::FullTransaction(Box::new(
-                                from_recovered(
-                                    tx.transaction.to_recovered_transaction(),
-                                    &tx_resp_builder,
-                                )
-                                .expect("fill should be infallible"),
-                            ))
+                        let stream = pubsub
+                            .full_pending_transaction_stream()
+                            .filter_map(|tx| {
+                                let tx_value = match from_recovered(
+                                tx.transaction.to_recovered_transaction(),
+                                &tx_resp_builder,
+                                ){
+                                    Ok(tx)  => Some(EthSubscriptionResult::FullTransaction(Box::new(tx))),
+                                    Err(err) => {
+                                        tracing::error!(target = "rpc", %err, "Failed to fill transaction with block context");
+                                        None
+                                    }
+                                };
+                                std::future::ready(tx_value)
```

Comment:
same here, this error should be propagated, or, sent as a message i.e. add new variant `SubscriptionMessageInner::Error(Box<dyn Error>)`. please open an issue.